Content.Server/Headset/HeadsetComponent.cs

```diff
@@ -55,6 +58,10 @@ public void Receive(string message, RadioChannelPrototype channel, EntityUid sou
 
             var playerChannel = actor.PlayerSession.ConnectedClient;
 
+            message = _chatSystem.TransformSpeech(source, message);
+            if (message.Length == 0)
+                return;
+
```

Comment:
This doesn't seem to be the place to transform speech, e.g. what about ghosts or handheld radios.

Reply:
All Handheld radios (If they are in range or something idk) call the *TrySendInGameICMessage* after being called by the same method through ` _listener.PingListeners(source, message, channel);`. 
Headsets also get called with that but they don't call the *TrySendGameICMessage* method because they only send a chat message.

Transforming the message before the call to ` _listener.PingListeners(source, message, channel);` would cause Handheld radios and other things that will listen to this in the future and then call the *TrySendInGameICMessage* method to transform the message twice.

If you really want to transform deadchat messages (They are a type of OOC message in the code) you would have to do that in the *SendDeadChat* method.